lib/loader/index.js

```diff
@@ -37,17 +37,24 @@ const ARRAY_SIZE = 16;
 const BIGINT = typeof BigUint64Array !== "undefined";
 const THIS = Symbol();
 
-const STRING_DECODE_THRESHOLD = 32;
-const decoder = new TextDecoder("utf-16le");
+const STRING_SMALLSIZE = 192; // break-even point in V8
+const STRING_CHUNKSIZE = 1024; // mitigate stack overflow
+const utf16 = new TextDecoder("utf-16le", { fatal: true }); // != wtf16
 
-/** Gets a string from an U32 and an U16 view on a memory. */
+/** Gets a string from memory. */
 function getStringImpl(buffer, ptr) {
-  const len = new Uint32Array(buffer)[ptr + SIZE_OFFSET >>> 2] >>> 1;
-  const arr = new Uint16Array(buffer, ptr, len);
-  if (len <= STRING_DECODE_THRESHOLD) {
-    return String.fromCharCode.apply(String, arr);
+  let len = new Uint32Array(buffer)[ptr + SIZE_OFFSET >>> 2] >>> 1;
+  const wtf16 = new Uint16Array(buffer, ptr, len);
+  if (len <= STRING_SMALLSIZE) return String.fromCharCode.apply(String, wtf16);
```

Comment:
It looks like make sense cache `String.fromCharCode.apply` to constant of function. like:
```ts
const fromCharCodes = String.fromCharCode.apply;
```

Reply:
First will give you "apply is called on undefined". Second may or may not make a difference after gzip, hm.

Reply:
> Second may or may not make a difference after gzip, hm.

Right, ok